src/polyswarm_api/__init__.py

```diff
@@ -556,16 +575,25 @@ def _fix_result(self, result):
 
         return results
 
-    async def download_file(self, h, destination_dir, with_metadata=False, hash_type='sha256'):
+    async def download_file(self, h, destination_dir, with_metadata=False, hash_type=None):
         """
         Download a file via the PS API
 
         :param h: Hash of the file you wish to download
         :param destination_dir: Directory you wish to save the file in
         :param with_metadata: Whether to save related file metadata into an associated JSON file
-        :param hash_type: Hash type [sha256|sha1|md5]
+        :param hash_type: Hash type [default:sha256|sha1|md5]
         :return: Dictionary containing path to the downloaded file if successful, error message if not
         """
+        if not hash_type or hash_type == 'sha1' or \
```

Comment:
Create `supported_hashtypes` attribute for this class, which is a list of hash types. Then just check `if not hash_type in self.supported_hashtypes`

Reply:
done